scopes/generator/generator/component-template.ts

```diff
@@ -1,23 +1,12 @@
 import { ComponentID } from '@teambit/component-id';
 
-export interface ComponentFile {
-  /**
-   * relative path of the file within the component.
-   */
-  relativePath: string;
-
-  /**
-   * file content
-   */
-  content: string;
-
-  /**
-   * whether this file will be tracked as the main file
-   */
-  isMain?: boolean;
-}
-
-export interface ComponentContext {
+/**
+ * ComponentContextOptions describes the shared properties between command and context
+ * related to component initialization and configuration.
+ *
+ * @template TComponentID - Represents the data type for the component ID properties.
+ */
+export interface ComponentTemplateOptions<TComponentID = ComponentID> {
```

Comment:
Please change this interface to have only the shared props of the two that implement it. 
No need for the generic here. Each can have its own (string/ComponentId).